src/external_printer.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(feature = "external_printer")]
+use crossbeam::channel::{bounded, Receiver, Sender};
+
+/// An ExternalPrinter allows to print messages of text while editing a line.
+/// The message is printed as a new line, the line-edit will continue below the
+/// output.
+#[cfg(feature = "external_printer")]
+#[derive(Debug, Clone)]
+pub struct ExternalPrinter {
+    sender: Sender<String>,
+    receiver: Receiver<String>,
```

Comment:
I forgot that the crossbeam channel by default is MPMC compared to the borked `std::mpsc`. Just cloning everything certainly makes the implementation simpler (while allowing some shenanigans like draining the channel from a different thread.)